### ISSUE TITLE
Integrated Console - Write-Progress messes things up

### ISSUE THREAD

**Issue by Tiberriver256** [PowerShell/vscode-powershell#3423]
### Issue Description

Running a script with `Write-Progress` screws up the console.

#### Example Script

```pwsh
function MyAwesomeScript {

	[CmdletBinding()]
	param (
		[Parameter()]
		[string]
		$Name
	)

	$WelcomeMessage = @();

	for ($i = 0; $i -lt 50; $i++) {
		Write-Progress -Activity "Getting ready to greet you!" -PercentComplete ($i/50*100)
		$WelcomeMessage += "Hello $Name"
	}

	$WelcomeMessage
}

MyAwesomeScript -Name Micah
```

#### Expected Behaviour

Should look like this (example taken from pwsh running in VSCode):

![image](https://user-images.githubusercontent.com/6989492/122449273-afa5cf00-cf73-11eb-8451-75926c9dabcf.png)

#### Actual Behaviour

Looks like this:

![image](https://user-images.githubusercontent.com/6989492/122449314-b9c7cd80-cf73-11eb-9fd6-1b4408b069ac.png)

The prompt is getting rendered towards the top instead of at the end of the content

**Comment by andschwa** [comment]
That is indeed a bug, thanks for reporting it. It's unfortunately a bit difficult to fix as its a custom PowerShell host. We will look into this.

**Comment by andschwa** [comment]
@rjmholt thinks this a race condition between `Write-Progress` and the printing of the prompt. This may be fixed by https://github.com/PowerShell/PowerShellEditorServices/issues/1295

**Comment by Tiberriver256** [comment]
🤞

**Comment by Tiberriver256** [comment]
Thanks for reaching out. Looks like it's still an issue in the preview version.

![image](https://user-images.githubusercontent.com/6989492/139504844-1e580f9e-af13-4b8a-a5c2-f980949a300c.png)

**Comment by StevenBucher98** [comment]
Thanks for checking @Tiberriver256! We will leave this issue open and continue to track it.

**Comment by SeeminglyScience** [comment]
A couple of things missing from the old host implementation:

https://github.com/PowerShell/PowerShellEditorServices/blob/6b1e77d6223ff61c4af3cbeedf687e1f65f3ac18/src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/EditorServicesPSHostUserInterface.cs#L609-L622

Track current progress messages so they can be cleared up when a the top level pipeline completes.  Also a little bit under that it's hooking into an execution status event to call `ClearProgress` after pipeline completion.

---

Since it's not very obvious why you'd need to do that, I'll copy and paste an explanation from when I was [suggesting it as the fix](https://github.com/PowerShell/vscode-powershell/issues/140#issuecomment-467279947):

> I still think the best bet is to reuse the existing `ConsoleHost`.  I tested it, it *does* work.  The problem is that since `ConsoleHost` isn't the one creating pipelines, it isn't cleaning up progress messages after each pipeline invocation.  There's a few options
> 
> 1. Keep track of the progress messages we send to `ConsoleHost` and send clean up progress messages for any that the user did not.  This would be my suggestion (assuming it works)

(it does)

**Comment by andschwa** [comment]
<img width="1076" alt="Screen Shot 2022-02-18 at 10 47 41 AM" src="https://user-images.githubusercontent.com/2226434/154744026-748d2c9f-2c87-4cad-9b4a-457cdfbfa5e5.png">

**Comment by andschwa** [comment]
The remaining fix for this is out in [PowerShell Preview v2022.2.2](https://marketplace.visualstudio.com/items?itemName=ms-vscode.powershell-preview)!
